src/framework/mlt_frame.c

```diff
@@ -887,29 +882,7 @@ mlt_frame mlt_frame_clone( mlt_frame self, int is_deep )
 			memcpy( copy, data, size );
 			mlt_properties_set_data( new_props, "audio", copy, size, mlt_pool_release, NULL );
 		}
-		data = mlt_properties_get_data( properties, "image", &size );
-		if ( data )
-		{
-			int width = mlt_properties_get_int( properties, "width" );
-			int height = mlt_properties_get_int( properties, "height" );
-
-			if ( ! size )
-				size = mlt_image_format_size( mlt_properties_get_int( properties, "format" ),
-					width, height, NULL );
-			copy = mlt_pool_alloc( size );
-			memcpy( copy, data, size );
-			mlt_properties_set_data( new_props, "image", copy, size, mlt_pool_release, NULL );
-
-			data = mlt_properties_get_data( properties, "alpha", &size );
-			if ( data )
-			{
-				if ( ! size )
-					size = width * height;
-				copy = mlt_pool_alloc( size );
-				memcpy( copy, data, size );
-				mlt_properties_set_data( new_props, "alpha", copy, size, mlt_pool_release, NULL );
-			};
-		}
+		mlt_image_copy_deep( &self->image, &new_frame->image );
```

Comment:
This is a good demonstration of how a lot of duplicated code can be replaced with single function calls.